umbraco/Umbraco-CMS
# v9: Implement IOptionsMonitor or IOptionsSnapshot instead of IOptions

# Notes
-Implemented IOptionsMonitor instead of IOptions for Unique and singleston services, this should allow changes in your config while running the application, so you can update your config on the fly!
- Implemented IOptionsSnapshot instead of IOptions for scoped services, as this will cache the value per request

# How to test
- Use https://our.umbraco.com/documentation/Reference/V9-Config/ for a list of config options
- Try to breakpoint the classes where I've made the change, it helps changing the OnChange to be like:
```
config.OnChange(x =>
                {
                    _myConfig = x;
                });
```
So you can see whenever the settings are changed
